orderbook/src/api/get_fee_and_price.rs

```diff
@@ -0,0 +1,186 @@
+};
+use std::convert::Infallible;
+use std::sync::Arc;
+use warp::{hyper::StatusCode, reply, Filter, Rejection, Reply};
+
+#[derive(Deserialize)]
+struct Query {
+    sell_token: H160Wrapper,
+    buy_token: H160Wrapper,
+    #[serde(with = "u256_decimal")]
+    amount: U256,
+    kind: OrderKind,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Response {
+    expiration_date: DateTime<Utc>,
+    min_fee: U256,
+    #[serde(with = "serde_with::rust::display_fromstr")]
+    amount_after_fee: BigInt,
+}
+
+#[derive(Debug)]
+enum Error {
+    NotFound,
+    UnsupportedToken(H160),
+    Other(anyhow::Error),
+}
+
+impl From<MinFeeCalculationError> for Error {
+    fn from(other: MinFeeCalculationError) -> Self {
+        match other {
+            MinFeeCalculationError::NotFound => Error::NotFound,
+            MinFeeCalculationError::UnsupportedToken(token) => Error::UnsupportedToken(token),
+            MinFeeCalculationError::Other(error) => Error::Other(error),
+        }
+    }
+}
+
+impl From<PriceEstimationError> for Error {
+    fn from(other: PriceEstimationError) -> Self {
+        match other {
+            PriceEstimationError::UnsupportedToken(token) => Error::UnsupportedToken(token),
+            PriceEstimationError::Other(error) => Error::Other(error),
+        }
+    }
+}
+
+fn request() -> impl Filter<Extract = (Query,), Error = Rejection> + Clone {
+    warp::path!("fee_and_price")
+        .and(warp::get())
+        .and(warp::query::<Query>())
+}
+
+fn response(result: Result<Response, Error>) -> impl Reply {
+    match result {
+        Ok(response) => reply::with_status(reply::json(&response), StatusCode::OK),
+        Err(Error::NotFound) => reply::with_status(
+            super::error("NotFound", "Token was not found"),
+            StatusCode::NOT_FOUND,
+        ),
+        Err(Error::UnsupportedToken(token)) => reply::with_status(
+            super::error("UnsupportedToken", format!("Token address {:?}", token)),
+            StatusCode::BAD_REQUEST,
+        ),
+        Err(Error::Other(err)) => {
+            tracing::error!(?err, "get_fee_and_price error");
+            reply::with_status(super::internal_error(), StatusCode::INTERNAL_SERVER_ERROR)
+        }
+    }
+}
+
+async fn calculate(
+    fee_calculator: Arc<dyn MinFeeCalculating>,
+    price_estimator: Arc<dyn PriceEstimating>,
+    query: Query,
+) -> Result<Response, Error> {
+    let (min_fee, expiration_date) = fee_calculator
+        .min_fee(query.sell_token.0, None, None, None)
+        .await?;
+    let price = price_estimator
+        .estimate_price(
+            query.sell_token.0,
+            query.buy_token.0,
+            query.amount,
+            query.kind,
+        )
+        .await?;
+    let amount_after_fee = match query.kind {
+        OrderKind::Buy => query.amount.to_big_rational() * price,
+        OrderKind::Sell => query.amount.saturating_sub(min_fee).to_big_rational() / price,
```

Comment:
Maybe to be more accurate we can do the subtraction before we estimate the price (but you also mentioned correctness is not the primary goal of this PR)?